third_party/amd/lib/TritonAMDGPUTransforms/BlockPingpong.cpp

```diff
@@ -364,17 +419,52 @@ void Pingponger::getDotPingponged() {
   //   GPU to hold all the data for the calculation. Such large tile size
   //   exceeds the amount of register GPU has so, we need to split the dot
   //   into several pieces.
+  //
+  // (3) Twp Dot-Memory (ping-pongx2) clusters
+  //  :Covers medium sized tile e.g., 256x128x64_FP16. Different tile size may
+  //  require different scheduling pattern because the loop consists of
+  //  different amount of memory transfer and dot operation. This scheduling
+  //  support the tile sizes not supported by above two methods.
+  //
+  // N.B., Tile size smaller than 128x128x64_FP16 is likely not compute-bound
+  // that pingpong scheduling doesn't help much.
+
+  auto dotType = dotOps[0].getType();
+  auto dotShape = dotType.getShape();
+  auto AType = dotOps[0].getA().getType();
+  auto AShape = AType.getShape();
+  auto elemWidth = AType.getElementType().getIntOrFloatBitWidth();
+  int64_t tileSize = dotShape[0] * dotShape[1] * AShape[1] * elemWidth;
+
+  const int64_t SmallTile = 16777216;  // e.g. 128x128x64x16bit
+  const int64_t MediumTile = 33554432; // SmallTile x 2
+  const int64_t LargeTile = 67108864;  // e.g. 256x256x64x16bit
+
+  auto encoding = cast<RankedTensorType>(AType).getEncoding();
+  auto srcEncoding = cast<ttg::DotOperandEncodingAttr>(encoding);
+  kWidth = srcEncoding.getKWidth();
+  auto mfmaEncoding =
+      dyn_cast<ttg::AMDMfmaEncodingAttr>(srcEncoding.getParent());
+  SmallVector<int64_t> intShape;
+  intShape.push_back(mfmaEncoding.getMDim());
+  intShape.push_back(mfmaEncoding.getNDim());
 
-  // TODO:
-  // - Add transformTwoPPClusters for the medium size tiles
-  // - Add definition of small/medium/large tile size considering data-type
-  //   so we can choose the transfrom per given tile size.
   if (numWarps == 4) { // pingpong between warps from different blocks
     // transfor a loop with small tile size
-    transformOnePPClusters(builder, loc);
+    if (tileSize == SmallTile)
```

Comment:
Does it also apply if tile size is `<= smallTile`?

Reply:
No harm but it doesn't help either. Added additional comment saying it's no longer advantageous to use is for the smaller tile size.